### PR TITLE
feat(aws-sqs-pipes-stepfunctions): new construct

### PR DESCRIPTION
**Pull request by biffgaut**
First of a new set of constructs integrating two serverless resources via EventBridge pipes

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d66d62653c6eb9e5bc26e2571f2e081ea865f2a3
* Result: FAILED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0df9bc8d-dc32-494f-ae93-258f10d4004f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 77f7dc17693770721754af1ba4af2c2ee06e3115
* Result: FAILED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=92952c20-fffc-4268-9067-85712b83fe0d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_sqs_pipes_stepfunctions`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-sqs-pipes-stepfunctions`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.sqspipesstepfunctions`|
+
+## Overview
+This AWS Solutions Construct implements an AWS SQS queue whose messages are passed to an AWS Step Functions state machine by and Eventbridge pipe.
```
Comment (hayesry) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md:22]: This AWS Solutions Construct implements an Amazon SQS queue whose messages are passed to an AWS Step Functions state machine by an Amazon EventBridge pipe.

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+new SqsToPipesToStepfunctions(this, 'SqsToLambdaToStepfunctionsPattern', {
+  stateMachineProps: {
+    definition: startState
+  }
+});
+```
+
+Python
+``` python
+TBD
```
Comment (hayesry) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md:44]: Maybe switch "TBD" to "TBA" or "Coming soon" (nit)

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+### AWS Step Functions State Machine
+* Deploy Step Functions standard state machine
+* Create CloudWatch log group with /vendedlogs/ prefix in name
+* Deploy best practices CloudWatch Alarms for the Step Functions
+
+### AWS EventBridge Pipe
+* Pipe configured with and SQS queue source and state machine target
+* A least privilege IAM role assigned to the pipe to access the queue and state machine
+* CloudWatch logs set up at the 'TBD' level
```
[the remark this answers is not in the document]
Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md:99]: I'm meeting with the TFC next week and will get the best practice level at that time.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: b9cfad210bd563039f2d9209151dbb5ad7fa8805
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=54c2346d-205b-4b2e-903a-8255a1319582%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d428f98d9aba4d6150d9765c8f2b469d39452720
* Result: FAILED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=693ab982-92d3-4760-8ad9-c3477501c377%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: b7ea39bf6e03d1fe9ce3e538520d501a0474cf6e
* Result: FAILED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e865be17-12f8-41a3-a361-aeb2218746dc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 39366a912c5ce7a8a4b3ad4c214d5ab2a83d0f43
* Result: FAILED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=43a08a36-1a66-4ff6-ad4d-044475407c23%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 7c1dcf0cd10d16a52e58feedb340cb5b682350b2
* Result: FAILED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ae644a80-d412-4614-a2cc-ed334b4c1b4a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f2213e3fb35545c89404d9ea84b0fa2e10af38a5
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5d113c32-34ce-4995-982b-50989213e8bf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Python
+``` python
+Coming soon
+```
+
+Java
+``` java
+Coming soon
+```
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md:50]: `Coming soon`, to me, implies the construct is not available for those languages. If this is very-short lived and the examples are updated soon then its fine.

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+|pipeProps?|[ pipes.CfnPipeProps ](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_pipes.CfnPipeProps.html)|Optional customer provided settings for the EventBridge pipe. source, target, roleArn and enrichment settings are set by the construct and cannot be overriden here. The construct will generate default sourceParameters, targetParameters and logConfiguration (found [here](link)) that can be overriden by populating those values in these props. If the client wants to implement enrichment or a filter, this is where that information can be provided. Any other props can be freely overridden. If a client wants to set values such as batchSize, that can be done here in the sourceParameters property.|
+| enrichmentFunction | [lambda.Function ](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html) | Optional - Lambda function that the construct will configure to be called to enrich the message between source and target. The construct will configure the pipe IAM role to allow invoking the function (but will not affect the IArole assigned to the function). Specifying both this and enrichmentStateMachine is an error. Default - undefined |
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md:68]: `enrichmentFunction?` - missing the `?`

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+| enrichmentStateMachine | [sfn.StateMachine ](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions.StateMachine.html) | Optional - Step Functions state machine that the construct will configure to be called to enrich the message between source and target. The construct will configure the pipe IAM role to allow executing the state machine (but will not affect the IAM role assigned to the state machine). Specifying both this and enrichmentStateMachine is an error. Default - undefined |
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/README.md:69]: `enrichmentStateMachine?` - missing the `?`

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+   * default = true.
+   */
+  readonly deployDeadLetterQueue?: boolean,
+  /**
+   * Optional user-provided props to override the default props for the dead letter SQS queue.
+   */
+  readonly deadLetterQueueProps?: sqs.QueueProps,
+  /**
+   * The number of times a message can be unsuccessfully dequeued before being moved to the dead letter
+   * queue. Defaults to 15.
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts:76]: To me, 15 retries seems high for a default. Is this number coming from a best practice doc some where?

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+   * @param {SqsToPipesToStepfunctionsProps} props - user provided props for the construct.
+   * @access public
+   */
+  constructor(scope: Construct, id: string, props: SqsToPipesToStepfunctionsProps) {
+    super(scope, id);
+
+    defaults.CheckStateMachineProps(props);
+    defaults.CheckSqsProps(props);
+    defaults.CheckPipesProps(props);
+
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts:145]: Nice and clean :)

**Review by georgebearden — APPROVED** [review]
looks good to me

File: source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts (hunk 7 above)
Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts:76]: It does seem high, it was set as a default 4 years ago. But changing it at this point would alter running workloads so I think just documenting it is our best option.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: c4c695611ad1ce082fead036e6759afad1aa6c70
* Result: SUCCEEDED
* [Build Logs](https://ab0is2wdo4.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4dcde0a2-4b53-4e10-963d-771777d7e476%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by georgebearden — DISMISSED** [review]

**Comment by biffgaut** [comment]
Merging without approval for the last commit as it is just doc changes suggested in the final approval.

**Pull request merged**